src/sentry/search/events/builder/profile_functions.py

```diff
@@ -120,9 +120,13 @@ def time_column(self) -> SelectType:
     def process_results(self, results: Any) -> EventsResponse:
         # Calling `super().process_results(results)` on the timeseries data
         # mutates the data in such a way that breaks the zerofill later such
-        # as applying `transform_alias_to_input_format` setting.  So skip it.
+        # as applying `transform_alias_to_input_format` setting.  So only run
+        # it to get the correct meta.
         for row in results["data"]:
             self.process_profiling_function_columns(row)
+
+        results["meta"] = super().process_results(results)["meta"]
```

Comment:
I think this function mutates the data, I need to double check if this will break anything. I think I have a test for it but Im not 100% sure.

Reply:
Looks like it doesn't mutate the data but returns a copy, so this is safe. However, I'm inclined to think we should split `process_results` into a separate functions to process each part of the result to avoid this trick of processing everything then keep only the meta.

Reply:
I almost did that, but then it turned out that the meta processing and the results processing use a lot of the same data, so the splitting was harder than I thought.

It's still doable though. I can do it in a separate PR and merge it ahead of this one if you feel it's important

Reply:
It's not an issue right now but the reason we'd want to fix is this because this will lead to the meta + data for profile functions to differ and might cause subtle issues down the line. But feel free to merge this for now and we can look into it later.